### ISSUE TITLE
Expose all functionality required by pyQuil

### ISSUE THREAD

**Issue by kalzoo** [rigetti/qcs-sdk-rust#139]
Currently, this SDK is minimalist in its implementation of the execution workflow. As we consider #135 and https://github.com/rigetti/pyquil/issues/1467, we must identify and implement the entrypoints into this SDK which will be needed by pyQuil.

These include:

- [ ] Translate a program without executing it

**Comment by randall-fulton** [comment]
The approach as of now:
```
pub async fn compile(
    quil: &str,
    quantum_processor_id: &str,
    config: &Configuration,
) -> Result<String, Box<dyn std::error::Error>>

pub async fn translate(
    native_quil: &str,
    shots: u16,
    quantum_processor_id: &str,
    config: &Configuration,
) -> Result<TranslateNativeQuilToEncryptedBinaryResponse, translation::Error>

pub async fn submit(
    program: &str,
    patch_values: HashMap<Box<str>, Vec<f64>>,
    quantum_processor_id: &str,
    config: &Configuration,
) -> Result<String, runner::Error>

pub async fn retrieve_results(
    job_id: &str,
    quantum_processor_id: &str,
    config: &Configuration,
) -> Result<GetExecutionResultsResponse, runner::Error>
```

**Comment by randall-fulton** [comment]
The core functionality here was implemented as part of #145
